java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java

```diff
@@ -107,20 +108,42 @@ public void initializeChildrenFromFields(List<Field> children) {
 
   @Override
   public void loadFieldBuffers(ArrowFieldNode fieldNode, List<ArrowBuf> ownBuffers) {
-    // variable width values: truncate offset vector buffer to size (#1)
-    org.apache.arrow.vector.BaseDataValueVector.truncateBufferBasedOnSize(ownBuffers, 1, offsets.getBufferSizeFor(fieldNode.getLength() + 1));
-    BaseDataValueVector.load(fieldNode, getFieldInnerVectors(), ownBuffers);
+    if (ownBuffers.size() != 2) {
+      throw new IllegalArgumentException("Illegal buffer count, expected " + 2 + ", got: " + ownBuffers.size());
+    }
+
+    ArrowBuf bitBuffer = ownBuffers.get(0);
+    ArrowBuf offBuffer = ownBuffers.get(1);
+
+    validityBuffer.release();
+    validityBuffer = bitBuffer.retain(allocator);
+    offsetBuffer.release();
+    offsetBuffer = offBuffer.retain(allocator);
+
+    validityAllocationSizeInBytes = validityBuffer.capacity();
+    offsetAllocationSizeInBytes = offsetBuffer.capacity();
+
     lastSet = fieldNode.getLength();
+    valueCount = fieldNode.getLength();
   }
 
   @Override
   public List<ArrowBuf> getFieldBuffers() {
-    return BaseDataValueVector.unload(getFieldInnerVectors());
+    List<ArrowBuf> result = new ArrayList<>(2);
+    validityBuffer.readerIndex(0);
+    validityBuffer.writerIndex(BitVectorHelper.getValidityBufferSize(valueCount));
+    offsetBuffer.readerIndex(0);
+    offsetBuffer.writerIndex((valueCount + 1) * OFFSET_WIDTH);
+
+    result.add(validityBuffer);
+    result.add(offsetBuffer);
+
+    return result;
   }
 
   @Override
   public List<BufferBacked> getFieldInnerVectors() {
```

Comment:
We should probably have a central place to document all removed APIs

Reply:
Sure. I will mark them deprecated.